### DISCUSSION TITLE
Examples of Different Design Algorithms

### DISCUSSION THREAD

**Discussion by mitchute** [j-c-cook/ghedt#83] [General]
I've seen (and used) the `near_square` design algorithm. What others are available? (I'm not seeing docs posted, but I definitely could have missed them) I'm specifically interested in area-constrained situations like you show in the GIF in the README. Where would I get started with something like that?

**Comment by j-c-cook** [comment]
See the [Design](https://github.com/j-c-cook/ghedt/tree/main/ghedt/examples/Design) directory in `examples/`. The best area constrained cases that currently have examples are `find_bi_rectangle.py` and `find_bi_zoned_rectangle.py`. 

An example for the case in the README is going to be added in the next release. [Issue 77](https://github.com/j-c-cook/ghedt/issues/77) is currently open regarding the addition of functionality to handle polygonal outer constraints and no-go drilling zones. 

I'd recommend that you look into `find_bi_rectangle.py` for now. It handles rectangular land constraints.

**Reply by NicholasFry** [reply]
I just ran the `find_bi_rectangle.py` script and it works great. I am kind of curious if I am specifying simulation parameters correctly. I understand the length width for the land areas. Why does the borehole length have one value while I specify a `max_height` and `min_height` for the PLAT params? What is the difference between the specified borehole length H and the range for the PLAT params?

**Reply by j-c-cook** [reply]
Good question. The value of the `H` variable is technically arbitrary. If you set the variable to `H = 1` and run the example, you will see that the results are the same. The automated field selection procedure is going to find a field that contains a height closest to and no greater than `max_height`. The simulations performed to determine the selection are all done at `max_height`. 

See the "call stack" below:

https://github.com/j-c-cook/ghedt/blob/93d159c7e6b42e0583ff837da37f40437f73bc32/ghedt/examples/Design/find_bi_rectangle.py#L146-L149

https://github.com/j-c-cook/ghedt/blob/93d159c7e6b42e0583ff837da37f40437f73bc32/ghedt/examples/Design/find_bi_rectangle.py#L153

https://github.com/j-c-cook/ghedt/blob/93d159c7e6b42e0583ff837da37f40437f73bc32/ghedt/design.py#L104-L110

https://github.com/j-c-cook/ghedt/blob/93d159c7e6b42e0583ff837da37f40437f73bc32/ghedt/search_routines.py#L256

Notice here that the bisection routine utilizes `self.sim_params.max_Height` for its simulations. 

https://github.com/j-c-cook/ghedt/blob/93d159c7e6b42e0583ff837da37f40437f73bc32/ghedt/search_routines.py#L177-L178

**Reply by NicholasFry** [reply]
This is awesome. Thank you!
